src/routes/console/project-[project]/auth/security/updatePasswordDictionary.svelte

```diff
@@ -0,0 +1,68 @@
+<script lang="ts">
+    import { invalidate } from '$app/navigation';
+    import { Submit, trackError, trackEvent } from '$lib/actions/analytics';
+    import { CardGrid, Heading } from '$lib/components';
+    import { Dependencies } from '$lib/constants';
+    import { Button, Form, FormList, InputNumber, InputSwitch } from '$lib/elements/forms';
+    import { addNotification } from '$lib/stores/notifications';
+    import { sdkForConsole } from '$lib/stores/sdk';
+    import { project } from '../../store';
+
+    const projectId = $project.$id;
+    let passwordDictionary = $project.authPasswordDictionary ?? 0;
+
+    async function updatePasswordDictionary() {
+        try {
+            const path = '/projects/' + projectId + '/auth/password-dictionary';
+            const uri = new URL(sdkForConsole.client.config.endpoint + path);
+            await sdkForConsole.client.call(
+                'patch',
+                uri,
+                {
+                    'content-type': 'application/json'
+                },
+                {
+                    enabled: passwordDictionary
+                }
+            );
+            invalidate(Dependencies.PROJECT);
+            addNotification({
+                type: 'success',
+                message: 'Updated password dictionary check.'
+            });
+            trackEvent(Submit.AuthPasswordDictionaryUpdate);
+        } catch (error) {
+            addNotification({
+                type: 'error',
+                message: error.message
+            });
+            trackError(error, Submit.AuthPasswordDictionaryUpdate);
+        }
+    }
+</script>
+
+<Form on:submit={updatePasswordDictionary}>
+    <CardGrid>
+        <Heading tag="h2" size="6">Password Dictionary</Heading>
+        <svelte:fragment slot="aside">
+            <FormList>
+                <InputSwitch
+                    bind:value={passwordDictionary}
+                    id="passwordDictionary"
+                    label="Password Dictionary" />
+            </FormList>
+            <p class="text">
+                When enabled it checks the user's password against the dictionary of most commonly
+                used password. It uses <a
```

Comment:
This link should open on a new page :blush: 

Also, `10 K` looks weird maybe `10K` is better?

Reply:
updated

Reply:
@lohanidamodar the link still opens in the same page. We can add `target="_blank" rel="noopener noreferrer"` to open it in a new tab

Reply:
added